internal/helm/release/manager.go

```diff
@@ -125,15 +126,12 @@ func (m *manager) Sync(ctx context.Context) error {
 	m.deployedRelease = deployedRelease
 	m.isInstalled = true
 
-	// Get the next candidate release to determine if an upgrade is necessary.
-	candidateRelease, err := m.getCandidateRelease(m.namespace, m.releaseName, m.chart, m.values)
-	if err != nil {
-		return fmt.Errorf("failed to get candidate release: %w", err)
-	}
-	if deployedRelease.Manifest != candidateRelease.Manifest {
-		m.isUpgradeRequired = true
-	}
+	upgrade := m.namespace == deployedRelease.Namespace
```

Comment:
Can you add a comment and a test case for this?

Reply:
> Can you add a comment and a test case for this?

Yes. I need some time